src/main/java/seedu/address/logic/parser/FindTagCommandParser.java

```diff
@@ -0,0 +1,27 @@
+package seedu.address.logic.parser;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.FindTagCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new FindTagCommand object.
+ */
+public class FindTagCommandParser implements Parser<FindTagCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments.
+     * @param args the input arguments
+     * @return a FindTagCommand object containing the input string
+     * @throws ParseException if the user input is empty.
+     */
+    public FindTagCommand parse(String args) throws ParseException {
+        if (args.isEmpty() || args.equals(" ")) {
+            throw new ParseException(String.format(
+                    Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindTagCommand.MESSAGE_USAGE));
+        }
+
+        assert args.charAt(0) == ' ';
```

Comment:
Not sure what youre doing here, why are you checking if the first Char is empty?

Reply:
because of how AddressBookParser parses the user input args is guaranteed to start with " ". this assertion just verifies that this is always true.

Reply:
sure, got it!